### PR TITLE
ENH: (More) comprehensive AnnexRepo.fsck() (fixes gh-3514)

### PR DESCRIPTION
**Pull request by mih**
This merely lifts the implementation from https://github.com/datalad/git-annex-ria-remote/blob/09a3eec05d89e2bb26f7ea2038d3e1aedd1ffe48/ria_remote/tests/utils.py#L32
and adds some docs.

A smoke test will come, likely infused into existing tests. I have no immediate plans to test the more exotic parts (scheduled incremental, etc).

### CODE REVIEW

**Comment by yarikoptic** [comment]
cool, I didn't even know we interfaced `annex fsck` ;)  The one recentish fsck experience for me was fixing up permissions, but ATM git-annex doesn't do that: http://git-annex.branchable.com/bugs/fsck_does_not_detect__47__fix_some_key_directories_correctly/ and I am [yet to convince](http://git-annex.branchable.com/bugs/fsck_does_not_detect__47__fix_some_key_directories_correctly/#comment-c35ff44643391ce41cc9187c582f00e8)  @joeyh to let machines do heavy lifting in some of those cases ;)

**Comment by mih** [comment]
Again (unrelated) time out in one of the appveyor runs.

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -2785,8 +2785,59 @@ def default_backends(self):
...
+          local repository.
+        fast : bool
+          If True, typically means that no actual content is being verified,
+          but tests are limited to the presence of files.
+        incremental : bool or {'continue'} or SCHEDULE
+          If given, `fsck` is called with `--incremental`. If 'continue',
+          `fsck` is additionally called with `--more`, and any other argument
+          is given to `--incremental-schedule`.
+        limit : str or all
+          If the functor `all` is given, `fsck` is called with `--all`. Any
```
Comment (kyleam) [inline datalad/support/annexrepo.py]: I'm guessing "functor" is a typo and you're not expecting readers to brush up on their category theory.

Comment (mih) [reply datalad/support/annexrepo.py]: Not a typo, this is how we talked about it in PyMVPA -- but I guess it could just be "function" ;-)

Comment (mih) [reply datalad/support/annexrepo.py]: Done

**Comment by mih** [comment]
Thx for the feedback @kyleam. I have commented out the pieces that are less important.

**Comment by kyleam** [comment]
> I have commented out the pieces that are less important.

Thanks.  However, if you have even a slight preference for leaving them in, please go with that.  Either way is fine with me.

**Comment by mih** [comment]
No, I am fine with leaving this out for now. We already have enough code with functionality that is not used much.

**Pull request merged**